From c0d890e23c4a4b2091e4e3efa4042efa0dec51ce Mon Sep 17 00:00:00 2001
From: "Satyanarayana, Sandhya" <sandhya.satyanarayana@ti.com>
Date: Fri, 29 Jun 2012 10:15:07 +0530
Subject: [PATCH 267/609] ARM: OMAP: AM33XX: Fix bug related to mDDR mode
 clear for PM

Original commit: b044423f5dcdcf2438132e0e6a396bdaa9b7fd3b

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

During the DDR3 resume sequence, mDDR mode has to be
cleared and IOs should be set for DDR2 (STL mode).

This patch fixes the issue where mddr_sel field was not
being cleared correctly to take the IOs out of mDDR mode.

Signed-off-by: Satyanarayana, Sandhya <sandhya.satyanarayana@ti.com>
---
 arch/arm/mach-omap2/sleep33xx.S |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/sleep33xx.S b/arch/arm/mach-omap2/sleep33xx.S
index f543c18..ed2156c 100644
--- a/arch/arm/mach-omap2/sleep33xx.S
+++ b/arch/arm/mach-omap2/sleep33xx.S
@@ -341,7 +341,7 @@ wait_emif_enable3:
 	/* Take out IO of mDDR mode */
 	ldr	r0, virt_ddr_io_ctrl
 	ldr	r1, [r0]
-	bic	r1, r1, #28
+	bic	r1, r1, #(1 << 28)
 	str	r1, [r0]
 
 	/* Enable VTP */
@@ -493,7 +493,7 @@ ENTRY(am33xx_resume_from_deep_sleep)
 	/* Take out IO of mDDR mode */
 	ldr	r0, phys_ddr_io_ctrl
 	ldr	r1, [r0]
-	bic	r1, r1, #28
+	bic	r1, r1, #(1 << 28)
 	str	r1, [r0]
 
 config_vtp3:
-- 
1.7.5.4

